### PR TITLE
Adds Ship to Ship combat!

### PR DESCRIPTION
**Pull request by Vivalas**
Wew.

This isn't entirely complete, and there a couple things to add, like, 

- More ships
- Better ship AI (so they can move around, retreat, jump to the help of their allies)
- Shields (for player ship, enemy ships use shields)
- Admin commands for spawning / removing ships
- Pirate faction that is hostile to everyone and can use any ship
- Fix ship generation so the neutral ships can be used
- Allow NPC ships to fight each other
- And anything I forgot, plus lots of bugfixes, I'm assuming.

### CODE REVIEW

File: code/datums/ship.dm [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+/datum/ship/faction/syndicate
+	name = "Syndicate"
+	cname = "syndicate"
+	relations = list("ship"=0,"nanotrasen"=0,"solgov"=0)
+
+/datum/ship/faction/ship
+	name = "Ship"
+	cname = "ship"
+	relations = list("nanotrasen"=1,"syndicate"=0,"solgov"=-1)
```
Comment (monster860) [inline code/datums/ship.dm]: I thought solvog was supposed to be neutral and syndies were supposed to be "fuck you"

File: code/game/machinery/computer/ftl_scanner.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if(href_list["target"])
+		var/datum/ship/starship/S = href_list["target"]
+		target = locate(S)
+
+	attack_hand(usr)
+
+/obj/machinery/computer/ftl_scanner/proc/status_update(var/message,var/sound)
+	visible_message("\icon[src] [message]")
+	if(sound)
+		playsound(src.loc,sound,50,0)
```
Comment (monster860) [inline code/game/machinery/computer/ftl_scanner.dm]: No src. please

File: code/game/machinery/computer/ftl_weapons.dm [hunk 3]
```diff
@@ -12,6 +13,10 @@
 	controller = new(src)
 	spawn(5)
 		refresh_weapons()
+		for(var/obj/machinery/computer/C in area_contents(src.loc.loc))
```
Comment (monster860) [inline code/game/machinery/computer/ftl_weapons.dm]: no src. please

File: code/modules/power/phase_cannon.dm [hunk 4]
```diff
@@ -58,9 +58,9 @@
 
 /obj/machinery/power/shipweapon/proc/attempt_fire()
 	if(!can_fire())
-		return
+		return 0
 	cell.use(200)
-	
+
 	var/obj/item/projectile/A = PoolOrNew(projectile_type,src.loc)
```
Comment (monster860) [inline code/modules/power/phase_cannon.dm]: no src. please

**Comment by CrAzYPiLoT-SS13** [comment]
Travis does not approve.

**Comment by Vivalas** [comment]
I see the issue, unused variable, will fix that.

File: code/datums/ship.dm (hunk 1 above)
Comment (Vivalas) [inline code/datums/ship.dm]: 0 is hostile, 1 is allied, -1 is enemy. I changed what they mean and didn't change the comment.

**Comment by Vivalas** [comment]
D:

**Comment by monster860** [comment]
There's literally no reason for components and factions to be a subclass of ships. Instead, make them `/datum/ship_component` and `/datum/faction`

**Comment by Vivalas** [comment]
I guess I can do that, I just didn't want to clutter the object tree

**Comment by monster860** [comment]
I don't think anyone gives a shit about the object tree outside of `/obj`, `/turf`, `/area`, and `/mob` (Which is used in the map editor)

**Comment by monster860** [comment]
Awesome

**Pull request merged**